app/services/follow_service.rb

```diff
@@ -1,22 +1,28 @@
 # frozen_string_literal: true
 class FollowService
-  class << self 
+  class << self
     def follow(entity, user, reason)
 
       return unless user
 
       return if (reason == 'created' || reason == 'contributed') && !should_auto_follow(entity, user, reason)
 
       follow = Follow.where(followed: entity, user: user).first_or_create
-      follow.update_attribute('muted', false)
+      unless follow.update(muted: false)
+        raise follow.errors.full_messages.join("\n")
```

Comment:
@devvmh hows this then? because it should fail the job, the failed job will send an exception

Reply:
yeah seems good to me.

Do we want to fail the whole job? I think yes, for a simple solution. Something more complicated might throw the error but continue on somehow. I'm happy with simple for now.

What use case is this? It's following a map from inside a background job? I don't see in my head when that happens

Reply:
it's used by activerecord callbacks for `create` and `update` for maps and topics, related to the auto-follow features

Reply:
when do we create/update topics in a delayed job?

Reply:
we don't, but the callbacks are set up to occur async instead of sync

Reply:
huh. i see. i don't fully understand but that shouldn't block merging